tests/unit/test_thrift_backend.py

```diff
@@ -968,6 +968,42 @@ def test_handle_execute_response_sets_active_op_handle(self):
 
         self.assertEqual(mock_resp.operationHandle, mock_cursor.active_op_handle)
 
+    @patch("thrift.transport.THttpClient.THttpClient")
+    @patch("databricks.sql.thrift_backend._retry_policy", new_callable=retry_policy_factory)
+    def test_make_request_will_retry_GetOperationStatus(
+            self, mock_retry_policy, t_transport_class):
+       
+        import thrift
+        from databricks.sql.thrift_api.TCLIService import TCLIService
+        from databricks.sql.exc import RequestError
+        from databricks.sql.utils import NoRetryReason
+
+        t_transport_instance = t_transport_class.return_value
+        t_transport_instance.code = 200
+        t_transport_instance.headers = {"foo": "bar"}
```

Comment:
I might just miss it, but where do we make sure an error is raised? As that is the point of this test, I'd make that explicit.

Reply:
Good catch. I updated the test case in 6ed85df867b48905a8a22634ee307cb4b0084631

Technically, this fix does not require that `GetOperationStatus` raises an exception. Only that no `Retry-After` header is present in the response from HttpTransport. But since this change was initially prompted by errors due to connection timeouts it makes sense to introduce that as the mechanism by which the request fails.